tests/integration/cli/test_networks.py

```diff
@@ -93,29 +93,40 @@ def assert_rich_text(actual: str, expected: str):
     """
     The output from `rich` causes a bunch of extra spaces to
     appear at the end of each line. For easier testing, we remove those here.
+    Also, we ignore whether the expected line is at the end of in the middle
+    of the output to handle cases when the test-runner has additional plugins
+    installed.
     """
-    expected = expected.strip()
-    lines = actual.split("\n")
-    new_lines = []
-    for line in lines:
-        if line:
-            new_lines.append(line.rstrip())
+    expected_lines = [
+        x.replace("└", "").replace("├", "").replace("│", "").strip()
+        for x in expected.strip().split("\n")
+    ]
+    actual_lines = [
+        x.replace("└", "").replace("├", "").replace("│", "").strip()
+        for x in actual.strip().split("\n")
+    ]
 
-    actual = "\n".join(new_lines)
-    assert actual == expected
+    for expected_line in expected_lines:
+        assert expected_line in actual_lines
```

Comment:
Is this function intended to be an exact match or more like an `x in y` match? 

If exact, you'll need to check the length and order of the two lists, since something like this could currently pass:

```python
expected = """
One
Two
Three
"""

actual = """
Four
Two
One
Three
One
"""
```

Reply:
The length may be different, so it is more like `x in y`.
It is different when the test runner has additional plugins installed.
And the order is not super important.

What is important is this:

* Comment out a a setting in `ape_geth.provides.GethNetworkConfig` and then run the test. It show now fail! So we are ensuring the provider is registered to specific networks.
* Change the `default_provider` in `ape_ethereum.ecosystem.NetworkConfig` from `geth` to `test`. The tests should no fail! So we are ensuring the correct default provider is listed

There are more examples that this integration test still ensures, but that is just a couple.

Reply:
Gotcha, so we basically just care that an expected string (like a network name) shows up at least once in the actual output